sw/device/rom_exts/rom_ext_manifest_parser.c

```diff
@@ -0,0 +1,117 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/rom_exts/rom_ext_manifest_parser.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+// TODO - parser accesses should be done through mmio and memory libraries.
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/rom_exts/manifest.h"
+
+uint32_t rom_ext_identifier_get(mmio_region_t base_addr) {
+  return mmio_region_read32(base_addr, ROM_EXT_MANIFEST_IDENTIFIER_OFFSET);
+}
+
+bool rom_ext_signature_get(mmio_region_t base_addr, rom_ext_signature_t *dst) {
+  if (dst == NULL) {
+    return false;
+  }
+
+  memcpy(&dst->data[0], (const void *)base_addr.base,
```

Comment:
You're not using the Signature offset, and tbh we should be using the mmio version of memcpy!

Reply:
Done.